lib/Frontend/CompilerInvocation.cpp

```diff
@@ -627,6 +627,16 @@ void LangOptions::setCxxInteropFromArgs(ArgList &Args,
       cxxInteropCompatVersion =
           validateCxxInteropCompatibilityMode("swift-5.9").second;
   }
+
+  for (const Arg *A : Args.filtered(options::OPT_Xcc)) {
+    StringRef clangArg = A->getValue();
+    if (clangArg.consume_front("-stdlib")) {
+      clangArg.consume_front("=");
+      if (clangArg.equals("libc++"))
+        CXXStdlib = CXXStdlibKind::OverrideLibcxx;
+      // If the stdlib argument is invalid, Clang will emit an error.
+    }
+  }
```

Comment:
I wonder if we should just let clang parse these and then pull it from the clang importer.

Reply:
Yeah, I initially thought of doing that in a similar way to how `getLibStdCxxFileMapping` extracts the libstdc++ installation path. This solution seemed to require a decent amount of plumbing through ClangImporter to get the stdlib kind to LangOptions, but let me see if I can do it in a simpler way.

Reply:
The main issue with extracting the stdlib kind from Clang seems to be Windows support: Clang doesn't seem to be aware that the Microsoft standard library is its own thing. It assumes that the default on Windows is libstdc++. It also cannot discover libc++ on Windows: 

https://github.com/llvm/llvm-project/blob/68df06a0b2998765cb0a41353fcf0919bbf57ddb/clang/lib/Driver/ToolChains/MSVC.cpp#L776-L779

Reply:
We can map that as a workaround, but we should really try to map the runtime rather than claim "default". What if there is no default (e.g. android)?

Reply:
(I would consider libc++ to be the default on Android – see the thread below)

Reply:
Done, I removed the manual parsing logic and switched to extracting the C++ stdlib kind from Clang Driver.